File: JAVA21/begining/DSA in C/WEEK1/fibonacci.c
#include<stdio.h>
int main(){
    int n,f;
    int fibo(int);
    printf("enter the number:");
    scanf("%d",&n);
    f = fibo(n);
    printf("fibonaci series at %d position number is: %d.",n,f);
    scanf("%d",&f);
    return 0;
}
int fibo(int n){
    int a=0,b=1,c;
    if(n==0)
        return a;
    else{
        for (int i=2;i<=n;i++){
            c = a+b;
            a = b;
            b = c;
        }
        return b;
    }    
}

File: DP/fibonacci.java
public class fibonacci {
    public static int fibo(int n, int f[]) {
        if (n == 0 | n == 1) {
            return n;
        }
        if (f[n] != 0) {
            return f[n];
        }
        f[n] = fibo(n - 1, f) + fibo(n - 2, f);
        return f[n];
    }

    public static void main(String[] args) {
        int n = 6;
        int f[] = new int[n + 1];
        System.out.println(fibo(n, f));
    }
}
